upup/pkg/fi/nodeup/command.go

```diff
@@ -163,7 +164,9 @@ func (c *NodeUpCommand) Run(out io.Writer) error {
 			return fmt.Errorf("error parsing BootConfig config response: %v", err)
 		}
 		nodeupConfigHash = sha256.Sum256([]byte(nodeConfig.NodeupConfig))
-		nodeupConfig.CAs[fi.CertificateIDCA] = bootConfig.ConfigServer.CACertificates
+		if len(bootConfig.ConfigServer.Servers) > 0 {
```

Comment:
`nodeConfig` can only be non-nil if this condition is true, so there's no need to test it.